lib/worker.js

```diff
@@ -259,23 +287,51 @@ class AssetComputeWorker {
 
         // post-process and upload
         for (let rendition of this.renditions) {
+            setConsoleLogPrefix(`[rendition ${rendition.index}]`);
+            try {
 
-            rendition = await this.postProcess(rendition);
-            if (!rendition) {
-                continue;
-            }
+                rendition = await this.postProcess(rendition);
+                if (!rendition) {
+                    continue;
+                }
 
-            if (this.options.disableRenditionUpload) {
-                await this.renditionSuccess(rendition);
-            } else if (rendition.exists()) {
-                await this.upload(rendition);
-            } else {
-                console.error(`no rendition found at: ${rendition.path}`);
-                await this.renditionFailure(rendition, new GenericError(`No rendition generated for ${rendition.id()}`, `${this.actionName}_batchProcess_norendition`));
+                if (this.options.disableRenditionUpload) {
+                    await this.renditionSuccess(rendition);
+                } else if (rendition.exists()) {
+                    await this.upload(rendition);
+                } else {
+                    console.error(`no rendition found at: ${rendition.path}`);
+                    await this.renditionFailure(rendition, new GenericError(`No rendition generated for ${rendition.id()}`, `${this.actionName}_batchProcess_norendition`));
+                }
+            } finally {
+                setConsoleLogPrefix();
             }
         }
     }
 
+    async preparePostProcess(rendition) {
+        // post processing could ask for different instructions for the worker callback
+        const intermediateInstructions = await prepareImagePostProcess(rendition, this.options);
+        if (intermediateInstructions) {
+            console.log("original instructions:", rendition.instructionsForEvent());
+
+            rendition.changeInstructions(intermediateInstructions);
+        }
+    }
+
+    async shouldPostProcess(rendition) {
+        if (!rendition.postProcess) {
```

Comment:
I would add a getter to see if `_forceImagePostProcess` is true and use it here. Because that means we're going to post-process anyway, and I think that one should have highest prio since we set it in the beginning when we change instructions to use an intermediary rendition.

Reply:
Still the worker is in full control, if it says no postProcess, it should not happen at all. Think custom workers or flite.

Reply:
I'm not sure we can have the worker having full control. The moment we use an intermediate rendition and update instructions, if we are not doing post-processing, what guarantees we return the rendition in the requested format, and not something half-baked?
E.g. we request a webp rendition. No worker can generate a webp rendition directly yet, so we switch instructions and ask instead for a png rendition. The worker should not know about that, but the SDK does. So, to finish and return the expected rendition, aren't we forced to run post-processing to export in the proper expected format?
The way I see/understood this, the worker should be blind to the fact that we are asking it for an intermediate rendition. The worker just wouldn't have enough information to decide alone, here... Wouldn't it?

Reply:
`rendition.postProcess` really is a feature flag saying "I am ready for post processing as a feature". We used it to roll out the feature selectively to our ootb workers. Now that all of them support it, we could just remove it and default to true.

However, before that we have to handle 2 things:
1. we use the flag in flite as a dynamic flag (can be true or false for different activations). this is a special case that could maybe done separately / different flag (skipPostProcessing=true, etc.)
2. we must keep it turned OFF for custom workers, because the post processing logic should never run for them at this point. also we could use a different way to detect that situation reliable, but it should be one that works with existing custom workers, otherwise it would be a breaking change if we ask them to set some flag in their code and redeploy